kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java

```diff
@@ -246,33 +246,29 @@ public static void assertPartitionOffsetsEqual(List<PartitionOffset> a, List<Par
         valueDeserializerClassName, new Properties(), validateContents);
   }
 
-  public static <T> void testWithRetry(Supplier<Boolean> testCondition, String errorMessage) {
-    testWithRetry(
-        testCondition,
-        errorMessage,
-        DEFAULT_EXP_BACKOFF_RETRIES,
-        DEFAULT_INITIAL_BACKOFF);
+  public static void testWithRetry(Runnable assertions) {
+    testWithRetry(assertions, DEFAULT_EXP_BACKOFF_RETRIES, DEFAULT_INITIAL_BACKOFF);
   }
 
-  public static <T> void testWithRetry(
-      Supplier<Boolean> testCondition,
-      String errorMessage,
-      int numRetries,
-      Duration initialBackoff) {
+  public static void testWithRetry(Runnable assertions, int numRetries, Duration initialBackoff) {
     Duration backoff = initialBackoff;
-    for (int i = 0; i <= numRetries; i++) {
-      if (testCondition.get()) {
+    for (int i = 0;; i++) {
+      try {
+        assertions.run();
         return;
+      } catch (Throwable e) {
```

Comment:
super-nit: The typical naming would be `Throwable t`.

Reply:
Done.